shared/util/quit-helper.desktop.js

```diff
@@ -3,18 +3,21 @@ import {ipcRenderer, ipcMain, app} from 'electron'
 import {quit} from '../../desktop/app/ctl'
 import {hideDockIcon} from '../../desktop/app/dock-icon'
 
-export type Context = {type: 'uiWindow'} | {type: 'mainThread'} | {type: 'quitButton'}
+export type Context = 'uiWindow' | 'mainThread' | 'quitButton' | 'beforeQuit'
```

Comment:
Not sure about this, the reason I did that is so we can add stuff to them later if we needed to, this makes it a bit harder